src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/create/actions/ResourcesAllocationAction.java

```diff
@@ -198,7 +210,7 @@ private void createPaths(HaFlowCreateFsm stateMachine) throws UnroutableFlowExce
     }
 
     private void allocateMainPath(HaFlowCreateFsm stateMachine) throws UnroutableFlowException,
-            RecoverableException, ResourceAllocationException {
```

Comment:
This exception can be thrown by allocateHaFlowResources().

Reply:
ah, it is swallowed by lombok and idea and compiler doesn't see it. Why to do that? Now we need to infer the exception ourselves instead of rely on the automation

Reply:
I think it is not because of lombok. I think it is because of `doInTransaction`

Reply:
right, doInTransaction has this mechanism called `@SneakyThrows`

Reply:
I put the exceptions back